### PR TITLE
Drag and drop a11y

### PR DESCRIPTION
**Pull request by phytonmk**
## Description

I have fully rebuilt DragAndDrop component, deprecated old apis and added new ones. Also new component was made accessible. Also updated some side components to not break DragAndDrop a11y.

## Motivation and Context

Previous DragAndDrop implementation was horribly bad.

## How has this been tested?

I have added VO test and updated snapshot test.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by freeyoungstrong — APPROVED** [review]

File: semcore/drag-and-drop/src/DragAndDrop.tsx [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        rect &&
+        event.clientX > rect.x &&
+        event.clientX < rect.x + rect.width &&
+        event.clientY > rect.y &&
+        event.clientY < rect.y + rect.height,
+    );
+    if (itemIndex === this.state.dragging?.index) this.setState({ previewSwap: null });
+    else this.setState({ previewSwap: itemIndex });
+  };
+  getFakeChildren = (index) => {
```
Comment (ekrasnopolskaia) [inline semcore/drag-and-drop/src/DragAndDrop.tsx:112]: maybe getPreviewChildren or smth else?

Comment (ekrasnopolskaia) [reply semcore/drag-and-drop/src/DragAndDrop.tsx:112]: and type missed

File: semcore/drag-and-drop/src/DragAndDrop.tsx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+      return false;
+    } else if (event.code === 'Escape' && this.state.dragging) {
+      event.preventDefault();
+      event.stopPropagation();
+      const a11yHint = `Dragging was discarded`;
+      this.setState({ a11yHint, dragging: null, previewSwap: null, hideHoverEffect: true });
+      return false;
+    }
+  };
+  handleFocus = (index) => () => {
```
Comment (ekrasnopolskaia) [inline semcore/drag-and-drop/src/DragAndDrop.tsx:176]: type missed

File: website/docs/components/drag-and-drop/drag-and-drop-a11y.md [hunk 3]
```diff
@@ -5,12 +5,13 @@ fileSource: drag-and-drop
 
 @## Keyboard support
 
-| Key                                  | Function                                                              |
-| ------------------------------------ | --------------------------------------------------------------------- |
-| `Space`                              | Selects the focused object and starts dragging.                       |
-| Repeated pressing of `Space`         | Terminates dragging.                                                  |
-| `Up`, `Down`, `Left`, `Right` arrows | Move the selected object down/up and right/left in the dropping area. |
+| Key                      | Function                                      |
```
Comment (ekrasnopolskaia) [inline website/docs/components/drag-and-drop/drag-and-drop-a11y.md:8]: I propose to add more descriptions here regarding keyboard usage

**Review by ekrasnopolskaia — COMMENTED** [review]
Great job, thank you

**Pull request merged**